### PR TITLE
Add BankId prefix to GrandId things that handles BankId

### PR DESCRIPTION
**Pull request by PeterOrneholm**
As discussed in #46 we should rename some things for GrandId to explicitly state that they are used for the BankId part to support Siths in a future release.

### CODE REVIEW

**Comment by PeterOrneholm** [comment]
The PR is now updated.
The methods on the API are now splitted so that only those parameters / result relevant to the usecase are present. This will enable Siths to have a more explicit API.

The Handler is splitted into a base and a BankId implementation, making it easy to have a Siths implementation.

File: src/ActiveLogin.Authentication.GrandId.AspNetCore/GrandIdAuthenticationBuilderExtensions.cs [hunk 1]
```diff
@@ -4,7 +4,6 @@ public static class GrandIdAuthenticationBuilderExtensions
     {
         internal static IGrandIdAuthenticationBuilder AddDefaultServices(this IGrandIdAuthenticationBuilder builder)
```
Comment (martinsjolanders) [inline src/ActiveLogin.Authentication.GrandId.AspNetCore/GrandIdAuthenticationBuilderExtensions.cs]: AddDefaultServices doesn't seem to do anything, mayby remove?

**Comment by martinsjolanders** [comment]
Should change 
return string.Join("&", queryStringParams.Select(x => $"{x.Key}={Uri.EscapeDataString(x.Value)}"));
to
QueryBuilder queryBuilder = new QueryBuilder(queryStringParams);
return queryBuilder.ToQueryString().ToString();
at
https://github.com/ActiveLogin/ActiveLogin.Authentication/blob/06bf812e6cae79f9f68c614814a4d1d7a18a3a65/src/ActiveLogin.Authentication.BankId.AspNetCore/BankIdLauncher.cs#L59

File: src/ActiveLogin.Authentication.GrandId.AspNetCore/GrandIdAuthenticationDefaults.cs [hunk 2]
```diff
@@ -5,21 +5,22 @@ namespace ActiveLogin.Authentication.GrandId.AspNetCore
...
-        public const string ChooseDeviceDisplayName = "BankID";
-        public static readonly PathString ChooseDeviceCallpackPath = new PathString("/signin-grandid-choosedevice");
+        public const string BankIdOtherDeviceAuthenticationScheme = "grandid-bankid-otherdevice";
+        public const string BankIdOtherDeviceDisplayName = "BankID (Annan enhet)";
+        public static readonly PathString BankIdOtherDeviceCallpackPath = new PathString("/signin-grandid-bankid-otherdevice");
 
-        public const string IdentityProviderName = "GrandId";
-        public const string AuthenticationMethodName = "grandid";
+        public const string BankIdChooseDeviceAuthenticationScheme = "grandid-bankid-choosedevice";
+        public const string BankIdChooseDeviceDisplayName = "BankID";
```
Comment (martinsjolanders) [inline src/ActiveLogin.Authentication.GrandId.AspNetCore/GrandIdAuthenticationDefaults.cs]: Should be BankID (Välj enhet)

Comment (PeterOrneholm) [reply src/ActiveLogin.Authentication.GrandId.AspNetCore/GrandIdAuthenticationDefaults.cs]: Thanks for input!

I've been thinking about naming in that, but my conclusion is that you will probably wither display "BankID (Denna enhet)" + "BankID (Annan enhet)" or only display "BankID (Välj enhet)", no reason to display them all three at once. And therefore, if it's only one BankID option available, it makes more sense to only call it BankID as I see it. What do you think?

**Comment by PeterOrneholm** [comment]
> Should change:

Fixed!

**Pull request merged**